### PR TITLE
Use typed access to request parameters

### PR DESCRIPTION
**Pull request by Alkarex**
This was a big source of mixed datatypes in many places

Contributes to https://github.com/FreshRSS/FreshRSS/issues/4112

### CODE REVIEW

File: lib/Minz/Request.php [hunk 1]
```diff
@@ -35,6 +42,7 @@ public static function params() {
 	 * @param mixed $default default value, if no parameter is given
 	 * @param bool $specialchars special characters
 	 * @return mixed value of the parameter
+	 * @deprecated use typed versions instead
```
Comment (Alkarex) [inline lib/Minz/Request.php:45]: Making the previous method deprecated.
Still used in one location: `FreshRSS_entry_Controller->readAction()`

**Comment by Alkarex** [comment]
@ColonelMoutarde would you feel like proofreading a bit here?

**Comment by Alkarex** [comment]
Let's merge to avoid too many PR conflicts

File: app/Controllers/configureController.php [hunk 2]
```diff
@@ -188,8 +188,8 @@ public function shortcutAction(): void {
 		$this->view->list_keys = SHORTCUT_KEYS;
 
 		if (Minz_Request::isPost()) {
-			$shortcuts = Minz_Request::param('shortcuts');
-			if (false !== Minz_Request::param('load_default_shortcuts')) {
+			$shortcuts = Minz_Request::paramArray('shortcuts');
+			if (!Minz_Request::paramBoolean('load_default_shortcuts')) {
```
Comment (Alkarex) [inline app/Controllers/configureController.php:192]: Negation bug fixed in https://github.com/FreshRSS/FreshRSS/pull/5294

File: app/Controllers/configureController.php [hunk 3]
```diff
@@ -454,14 +453,14 @@ public function systemAction(): void {
 
 		if (Minz_Request::isPost()) {
 			$limits = FreshRSS_Context::$system_conf->limits;
-			$limits['max_registrations'] = Minz_Request::param('max-registrations', 1);
-			$limits['max_feeds'] = Minz_Request::param('max-feeds', 16384);
-			$limits['max_categories'] = Minz_Request::param('max-categories', 16384);
-			$limits['cookie_duration'] = Minz_Request::param('cookie-duration', FreshRSS_Auth::DEFAULT_COOKIE_DURATION);
+			$limits['max_registrations'] = Minz_Request::paramInt('max-registrations') ?: 1;
```
Comment (Inverle) [inline app/Controllers/configureController.php:456]: #7751

**Pull request merged**